client/src/Movies/Movie.js

```diff
@@ -10,59 +12,41 @@ export default class Movie extends Component {
   }
 
   componentDidMount() {
-    // change this line to grab the id passed on the URL
-    const id = 1;
-    this.fetchMovie(id);
-  }
-
-  fetchMovie = id => {
+    const {id} = this.props.match.params;
     axios
       .get(`http://localhost:5000/api/movies/${id}`)
       .then(response => {
         this.setState(() => ({ movie: response.data }));
       })
-      .catch(error => {
-        console.error(error);
-      });
-  };
+      .catch(error => console.error(error));
+  }
+
+  // fetchMovie = id => {
+  //   axios
+  //     .get(`http://localhost:5000/api/movies/${id}`)
+  //     .then(response => {
+  //       this.setState(() => ({ movie: response.data }));
+  //     })
+  //     .catch(error => console.error(error));
+  // };
+
   // Uncomment this code when you're ready for the stretch problems
   // componentWillReceiveProps(newProps){
   //   if(this.props.match.params.id !== newProps.match.params.id){
   //     this.fetchMovie(newProps.match.params.id);
   //   }
   // }
 
-  // saveMovie = () => {
-  //   const addToSavedList = this.props.addToSavedList;
-  //   addToSavedList(this.state.movie)
-  // }
-
-  render() {
-    if (!this.state.movie) {
-      return <div>Loading movie information...</div>;
-    }
+  saveMovie = () => {
+    // const addToSavedList = this.props.addToSavedList;
+    // addToSavedList(this.state.movie);
 
-    const { title, director, metascore, stars } = this.state.movie;
-    return (
-      <div className="save-wrapper">
-        <div className="movie-card">
-          <h2>{title}</h2>
-          <div className="movie-director">
-            Director: <em>{director}</em>
-          </div>
-          <div className="movie-metascore">
-            Metascore: <strong>{metascore}</strong>
-          </div>
-          <h3>Actors</h3>
+    this.props.addToSavedList(this.state.movie);
```

Comment:
the saveMovie function is unnecessarily clunky to start with, good choice to streamline it